{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import matplotlib\n",
    "%matplotlib inline\n",
    "import numpy as np\n",
    "from scipy import stats\n",
    "matplotlib.rcParams['svg.fonttype'] = 'none'\n",
    "matplotlib.rc('font', family='sans-serif')\n",
    "matplotlib.rc('font', serif='Arial')\n",
    "matplotlib.rc('text', usetex='false')\n",
    "matplotlib.rc('pdf', fonttype=42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## let's instead count all non-fixed variants in the data\n",
    "\n",
    "##a wrapper class to store a parsed line from a VCF file\n",
    "class Variant:\n",
    "    chrome = ''\n",
    "    pos = 0\n",
    "    line = ''\n",
    "    ##ref is in the 0 position, alts further\n",
    "    nuc_seqs = []\n",
    "    ##list of list of genotypes\n",
    "    gts = []\n",
    "    ##list of all GT quality scores\n",
    "    quals = []\n",
    "    ##reads supporting each allele\n",
    "    read_depth = []\n",
    "\n",
    "##takes a line from a VCF files and parses it\n",
    "def parse_vcf_list(vcf_list):\n",
    "    ##store the values in the Variant object\n",
    "    cur_variant = Variant()\n",
    "    cur_variant.chrome = vcf_list[0]\n",
    "    cur_variant.pos = int(vcf_list[1])\n",
    "\n",
    "    ##set the sequences for ref and alt, allowing for > 1 alt allele\n",
    "    cur_ref= [vcf_list[3]]\n",
    "    cur_alts = vcf_list[4].split(',')\n",
    "    cur_seqs =cur_ref + cur_alts\n",
    "    cur_variant.nuc_seqs = cur_seqs\n",
    "\n",
    "    ##process the raw genotypes\n",
    "    raw_gts = vcf_list[9:]\n",
    "    cur_gts =  map(lambda x: x.split(':')[0].split('/'),raw_gts)\n",
    "    #cur_gts =  map(lambda x: map(lambda y: int(y),x),cur_gts)\n",
    "    cur_variant.gts = cur_gts\n",
    "    cur_quals = []\n",
    "    cur_rd_list = []\n",
    "    for cur_raw_gt in raw_gts:\n",
    "        cur_qual = 0\n",
    "        cur_rd = []\n",
    "        split = cur_raw_gt.split(':')\n",
    "        if len(split) > 3:\n",
    "            if split[3] == '.':\n",
    "                cur_qual=0\n",
    "                \n",
    "            else:\n",
    "                cur_qual = int(split[3])\n",
    "                cur_rd = map(lambda x:int(x),split[1].split(','))\n",
    "        cur_quals.append(cur_qual)\n",
    "        cur_rd_list.append(cur_rd)\n",
    "    #cur_quals = int(map(lambda x: x.split(':')[3],raw_gts))\n",
    "    cur_variant.quals=cur_quals\n",
    "    cur_variant.read_depth = cur_rd_list\n",
    "    return cur_variant\n",
    "\n",
    "##opens and parses a VCF file line by line\n",
    "##yields a Variant object for each line\n",
    "def VCF_parser(vcf_file):\n",
    "    \n",
    "\n",
    "\n",
    "    f = open(vcf_file,'r')\n",
    "    \n",
    "    for line in f:\n",
    "        ##chomp dat header\n",
    "        if line.startswith('#'):\n",
    "            continue\n",
    "            \n",
    "        line = line.strip()\n",
    "        split = line.split('\\t')\n",
    "        \n",
    "        cur_variant = parse_vcf_list(split)\n",
    "        cur_variant.line = line\n",
    "        \n",
    "        yield cur_variant\n",
    "        \n",
    "\n",
    "##counts non-fixed variants\n",
    "def count_unfixed_variants(invcf,MIN_GT=0):     \n",
    "    \n",
    "    unique_vars = 0\n",
    "\n",
    "    \n",
    "    for cur_variant in VCF_parser(invcf):\n",
    "        ##keep track of variants we have seen, if > 1, then the variant isn't fixed\n",
    "        seen_vars = set()\n",
    "        for i,genotype in enumerate(cur_variant.gts):\n",
    "            for allele in genotype:\n",
    "                cur_qual = cur_variant.quals[i]\n",
    "                ##only continue if the variant is of sufficient quality\n",
    "                if cur_qual < MIN_GT:\n",
    "                    continue\n",
    "                \n",
    "                ##if the allele isn't a digit, it is missing, ignore it\n",
    "                if not allele.isdigit():\n",
    "                    continue\n",
    "                \n",
    "                \n",
    "                seen_vars.add(allele)\n",
    "        \n",
    "        ##count all unfixed alleles (>1 allele can be unfixed)\n",
    "        ##can't be less than 0\n",
    "        cur_new_vars = max(len(seen_vars) - 1,0)\n",
    "        if cur_new_vars >= 1:\n",
    "            pass\n",
    "            #print cur_variant.nuc_seqs\n",
    "            #print cur_variant.gts\n",
    "            #print cur_variant.chrome,cur_variant.pos\n",
    "        unique_vars += cur_new_vars\n",
    "    \n",
    "    return unique_vars\n",
    "        \n",
    "DNA_sample_dict = {\n",
    "    'AG07A'   : 'CERC',\n",
    "    'AG07B'   : 'SAPC',\n",
    "    'AG07C'   : 'GARC',\n",
    "    'AG07D'   : 'BEPA;dad(BEPAxLITC)',\n",
    "    'AG07E'   : 'LITC;mom(BEPAxLITC)',\n",
    "    'AG07F'   : 'FRI;dad(FRIxLITC)',\n",
    "    'AG08A'   : 'FTC;dad(FTCxLITC)',\n",
    "    'AG08B'   : 'LITC;mom(mom FTCxLITC)',\n",
    "    'AG08C'   : 'LITC;dad(LITCxLITC 3)',\n",
    "    'AG08D'   : 'LITC;mom(PAXBxLITC 28)',\n",
    "    'AG08E'   : 'LITC;mom(PAXBxLITC 29)',\n",
    "    'AG08F'   : 'PAXB;dad(PAXBxLITC 28/29)',\n",
    "    'AMG'     : 'FTC;mom(BigRev) ',\n",
    "    'AMG01'   : 'LITC;dad(BigRev)',\n",
    "    'JCH002A' : 'PAXB;dad(PAXBXRABS)',\n",
    "    'JCH002B' : 'PAXB;dad(PAXBXJAMA-ANTHONY)',\n",
    "    'JCH002C' : 'PAXB;dad(PAXBxJAMA-LYNDA)',\n",
    "    'JCH002D' : 'PAXB;dad(WV)',\n",
    "    'JCH002E' : 'PAXB;dad(JCH_ASE)',\n",
    "    'JCH002F' : 'RABS;mom(JCH_DNaseI)',\n",
    "    'JCH006A' : 'CERC;JCH053113-1M)',\n",
    "    'JCH006B' : 'CERC;AML110812-1M',\n",
    "    'JCH006C' : 'JAMA;PSL040511-2F',\n",
    "    'JCH006D' : 'Hutu F3',\n",
    "    'JCH006E' : 'CCD 2007',\n",
    "    'JCH007A' : 'LITC LCM || benthic',\n",
    "    'JCH007B' : 'EnosB || benthic',\n",
    "    'JCH007C' : 'PriestB || benthic',\n",
    "    'JCH007D' : 'PaxtonB || benthic',\n",
    "    'PCAG5'   : 'PAXBxLITC;F2 (21=D)',\n",
    "    'PCAG6'   : 'PAXBxLITC;F2 (21=B)',\n",
    "    'PCAG7'   : 'CF_FTC;mom',\n",
    "    'PCAG8'   : 'CF_LITC;dad',\n",
    "    'JCH009G' : 'RABS;mom (CERC x RABS F2 Mom)'\n",
    "}\n",
    "\n",
    "PHYLIP_sample_dict = {\n",
    "    'AG07A'   : 'CERC-wd454',\n",
    "    'AG07B'   : 'SAPC-wd448',\n",
    "    'AG07C'   : 'GARC-wd711',\n",
    "    'AG07D'   : 'BEPA-wdBxL',\n",
    "    'AG07E'   : 'LITC-wmBxL',\n",
    "    'AG07F'   : 'FRI-wdFRxL',\n",
    "    'AG08A'   : 'FTC-wdFTxL',\n",
    "    'AG08B'   : 'LITC-mFTxL',\n",
    "    'AG08C'   : 'LITC-wdLC3',\n",
    "    'AG08D'   : 'LITC-wmL28',\n",
    "    'AG08E'   : 'LITC-wmL29',\n",
    "    'AG08F'   : 'PAXB-wdPxL',\n",
    "    'AMG'     : 'FTC-lmBRev',\n",
    "    'AMG01'   : 'LITC-ldBRv',\n",
    "    'JCH002A' : 'PAXB-ldPxR',\n",
    "    'JCH002B' : 'PAXB-dPxJa',\n",
    "    'JCH002C' : 'PAXB-dPxJl',\n",
    "    'JCH002D' : 'PAXB-dWVir',\n",
    "    'JCH002E' : 'PAXB-dJASE',\n",
    "    'JCH002F' : 'RABS-mDNas',\n",
    "    'JCH006A' : 'CERC-dCASE',\n",
    "    'JCH006B' : 'CERC-dCxRB',\n",
    "    'JCH006C' : 'JAMA-mPxJl',\n",
    "    'JCH006D' : 'Hutu F3   ',\n",
    "    'JCH006E' : 'CCD 2007  ',\n",
    "    'JCH007A' : 'LITC-wmPB ',\n",
    "    'JCH007B' : 'EnosB-wdPB',\n",
    "    'JCH007C' : 'PriestB-wd',\n",
    "    'JCH007D' : 'PaxtonB-wd',\n",
    "    'PCAG5'   : 'PxLF2;21=D',\n",
    "    'PCAG6'   : 'PxLF2;21=B',\n",
    "    'PCAG7'   : 'CF_FTC;mom',\n",
    "    'PCAG8'   : 'CF_LITCdad',\n",
    "    'JCH009G' : 'RABS-lmCxR'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n['AG07A', 'AG07D', 'AG07E', 'AG07F', 'AG08A', 'AG08B', 'AG08D', 'AG08E', 'AG08F', 'AMG', 'AMG01', 'JCH002A', 'JCH002B', 'JCH002C', 'JCH002D', 'JCH006B', 'JCH006C', 'JCH006D', 'JCH006E', 'JCH007A', 'JCH007B', 'JCH007C', 'JCH007D']\\n[ 8  7  7  6  8  9 14 13  9 60 59  4  6  8  7  5  8  3  5 11 11 12  9]\\n[ 31  25  26  20  30  32  46  45  32 109 108  13  21  25  22  18  26  11\\n  18  37  36  38  30]\\n1092385 397985 6308942 857751 2044600\""
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##filters invcf, gets rid of variants where less that PROP_PASS of samples have MIN_RD and MIN_GQ requirements\n",
    "##Will keep header lines if HEADER=True\n",
    "def filter_vcf(invcf,outvcf,HEADER=True,MIN_RD=3,MIN_GQ=5,PROP_FAIL=0,ONLY_SNPs=True,DIALLELE=True):\n",
    "    \n",
    "    f = open(invcf,'r')\n",
    "    w = open(outvcf,'w')\n",
    "    genome_ids = []\n",
    "    if HEADER:\n",
    "        for line in f:\n",
    "            if line.startswith('##'):\n",
    "                w.write(line)\n",
    "            elif line.startswith('#'):\n",
    "                w.write(line)\n",
    "                genome_ids = line.strip().split()[9:]\n",
    "            else:\n",
    "                break\n",
    "        f.close()      \n",
    "        f = open(invcf,'r')\n",
    "    print genome_ids  \n",
    "    low_qual = 0\n",
    "    low_rd = 0\n",
    "    both = 0\n",
    "    hq = 0\n",
    "    total = 0\n",
    "    indel = 0\n",
    "    total_rd = []\n",
    "    total_qual = []\n",
    "    for cur_variant in VCF_parser(invcf):\n",
    "        ##remove chrUn and chrXIX (the X chromosome)\n",
    "        if cur_variant.chrome == 'chrUn' or cur_variant.chrome == 'chrXIX':\n",
    "            continue\n",
    "        ##We want total read depth, not any specific variant\n",
    "        cur_variant.read_depth = map(sum,cur_variant.read_depth) \n",
    "        \n",
    "        \n",
    "         \n",
    "        \n",
    "        read_depth_array = np.array(cur_variant.read_depth)\n",
    "        if total_rd == []:\n",
    "            total_rd = read_depth_array\n",
    "        else:\n",
    "            total_rd = total_rd + read_depth_array\n",
    "        qual_array = np.array(cur_variant.quals)\n",
    "        if total_qual == []:\n",
    "            total_qual = qual_array\n",
    "        else:\n",
    "            total_qual = total_qual + qual_array\n",
    "        \n",
    "        low_rd_sample_num =  len(read_depth_array[read_depth_array < MIN_RD])\n",
    "        low_qual_sample_num =  len(qual_array[qual_array < MIN_GQ]) \n",
    "        \n",
    "        ##minimum number of low quality samples per variant\n",
    "        min_low_num = len(read_depth_array) * PROP_FAIL\n",
    "        \n",
    "        \n",
    "        total =  hq + both + low_rd + low_qual \n",
    "        \n",
    "        \n",
    "        \n",
    "        IS_SNP=True\n",
    "        \n",
    "        for genotype in cur_variant.nuc_seqs:\n",
    "            if len(genotype) > 1:\n",
    "                IS_SNP = False\n",
    "                \n",
    "        if not IS_SNP:\n",
    "            indel +=1\n",
    "            if ONLY_SNPs:\n",
    "                continue\n",
    "                \n",
    "        IS_MULTI = False\n",
    "        for genotype in cur_variant.gts:\n",
    "            for allele in genotype:\n",
    "                if allele != '.' and allele != '0' and allele != '1':\n",
    "                    IS_MULTI = True\n",
    "                    \n",
    "        if DIALLELE and IS_MULTI:\n",
    "            continue\n",
    "        \n",
    "        '''if total > 0 and (total % 100000) == 0:\n",
    "            print cur_variant.chrome,low_rd,low_qual,both,hq,indel\n",
    "            print total_rd/total\n",
    "            print total_qual/total\n",
    "        '''\n",
    "        if low_rd_sample_num > min_low_num and low_qual_sample_num > min_low_num:\n",
    "            both += 1\n",
    "            continue\n",
    "        elif low_rd_sample_num >min_low_num:\n",
    "            low_rd += 1\n",
    "            continue\n",
    "        elif low_qual_sample_num > min_low_num:\n",
    "            low_qual += 1\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            hq +=1\n",
    "            w.write(cur_variant.line + '\\n')\n",
    "            \n",
    "        total =  hq + both + low_rd + low_qual\n",
    "        \n",
    "        \n",
    "                    \n",
    "        \n",
    "            \n",
    "            \n",
    "        \n",
    "    print low_rd,low_qual,both,hq,indel\n",
    "    print total_rd/total\n",
    "    print total_qual/total\n",
    "        \n",
    "        \n",
    "        \n",
    "    f.close()\n",
    "    w.flush()\n",
    "    w.close()\n",
    "#filter_vcf('MOAV_071817_tested_haplotype.vcf','MOAV_071817_tested_filtered_haplotype.vcf')    \n",
    "'''\n",
    "['AG07A', 'AG07D', 'AG07E', 'AG07F', 'AG08A', 'AG08B', 'AG08D', 'AG08E', 'AG08F', 'AMG', 'AMG01', 'JCH002A', 'JCH002B', 'JCH002C', 'JCH002D', 'JCH006B', 'JCH006C', 'JCH006D', 'JCH006E', 'JCH007A', 'JCH007B', 'JCH007C', 'JCH007D']\n",
    "[ 8  7  7  6  8  9 14 13  9 60 59  4  6  8  7  5  8  3  5 11 11 12  9]\n",
    "[ 31  25  26  20  30  32  46  45  32 109 108  13  21  25  22  18  26  11\n",
    "  18  37  36  38  30]\n",
    "1092385 397985 6308942 857751 2044600'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['AG07A', 'AG07D', 'AG07E', 'AG07F', 'AG08A', 'AG08B', 'AG08D', 'AG08E', 'AG08F', 'AMG', 'AMG01', 'JCH002A', 'JCH002B', 'JCH002C', 'JCH002D', 'JCH006B', 'JCH006C', 'JCH006D', 'JCH006E', 'JCH007A', 'JCH007B', 'JCH007C', 'JCH007D', 'JCH009G'] ['CERC-wd454', 'BEPA-wdBxL', 'LITC-wmBxL', 'FRI-wdFRxL', 'FTC-wdFTxL', 'LITC-mFTxL', 'LITC-wmL28', 'LITC-wmL29', 'PAXB-wdPxL', 'FTC-lmBRev', 'LITC-ldBRv', 'PAXB-ldPxR', 'PAXB-dPxJa', 'PAXB-dPxJl', 'PAXB-dWVir', 'CERC-dCxRB', 'JAMA-mPxJl', 'Hutu F3   ', 'CCD 2007  ', 'LITC-wmPB ', 'EnosB-wdPB', 'PriestB-wd', 'PaxtonB-wd', 'RABS-lmCxR']\n",
      "['AG07A', 'AG07D', 'AG07E', 'AG07F', 'AG08A', 'AG08B', 'AG08D', 'AG08E', 'AG08F', 'AMG', 'AMG01', 'JCH002A', 'JCH002B', 'JCH002C', 'JCH002D', 'JCH006B', 'JCH006C', 'JCH006D', 'JCH006E', 'JCH007A', 'JCH007B', 'JCH007C', 'JCH007D', 'JCH009G'] ['CERC-wd454', 'BEPA-wdBxL', 'LITC-wmBxL', 'FRI-wdFRxL', 'FTC-wdFTxL', 'LITC-mFTxL', 'LITC-wmL28', 'LITC-wmL29', 'PAXB-wdPxL', 'FTC-lmBRev', 'LITC-ldBRv', 'PAXB-ldPxR', 'PAXB-dPxJa', 'PAXB-dPxJl', 'PAXB-dWVir', 'CERC-dCxRB', 'JAMA-mPxJl', 'Hutu F3   ', 'CCD 2007  ', 'LITC-wmPB ', 'EnosB-wdPB', 'PriestB-wd', 'PaxtonB-wd', 'RABS-lmCxR']\n"
     ]
    }
   ],
   "source": [
    "##converts two nucleotides into a single letter (e.g. A, C -> M)\n",
    "##This seems like it will work for DNAML\n",
    "def convertDiNT(nuc1,nuc2):\n",
    "    \n",
    "    NTs = [nuc1,nuc2]\n",
    "    \n",
    "    if 'A' in NTs and 'C' in NTs:\n",
    "        return 'M'\n",
    "    if 'A' in NTs and 'G' in NTs:\n",
    "        return 'R'\n",
    "    if 'A' in NTs and 'T' in NTs:\n",
    "        return 'W'\n",
    "    if 'C' in NTs and 'G' in NTs:\n",
    "        return 'S'\n",
    "    if 'C' in NTs and 'T' in NTs:\n",
    "        return 'Y'\n",
    "    if 'G' in NTs and 'T' in NTs:\n",
    "        return 'K'\n",
    "    else:\n",
    "        return nuc1\n",
    "\n",
    "\n",
    "##makes a VCF file into a tsv matrix, usable for PCA\n",
    "def vcf_to_gts(invcf,outgt,CHROME='',START=0,STOP=0):\n",
    "    \n",
    "    w = open(outgt,'w')\n",
    "    f = open(invcf,'r')\n",
    "    \n",
    "    for line in f:\n",
    "        line = line.strip()\n",
    "        if line.startswith('##'):\n",
    "            continue\n",
    "        split = line.split()\n",
    "        ids = split[9:]\n",
    "        names = map(PHYLIP_sample_dict.get,ids,ids)\n",
    "        break\n",
    "    print ids,names\n",
    "    f.close()\n",
    "    header_list = ['chr:pos'] + names\n",
    "    w.write('\\t'.join(header_list) + '\\n')\n",
    "    \n",
    "    for cur_variant in VCF_parser(invcf):\n",
    "        \n",
    "        if CHROME and START and STOP:\n",
    "            \n",
    "            if cur_variant.chrome != CHROME:\n",
    "                continue\n",
    "            if cur_variant.pos < START or cur_variant.pos > STOP:\n",
    "                continue\n",
    "\n",
    "        ##collapse scores into single number\n",
    "        cur_gt_scores = map(lambda x: str(int(x[0]) + int(x[1])),cur_variant.gts)\n",
    "        \n",
    "        cur_chrpos = cur_variant.chrome + ':' + str(cur_variant.pos)\n",
    "        \n",
    "        new_list_list = [cur_chrpos] + cur_gt_scores\n",
    "        w.write('\\t'.join(new_list_list) + '\\n')\n",
    "    w.flush()\n",
    "    w.close()\n",
    "        \n",
    "        \n",
    "        \n",
    "    \n",
    "\n",
    "##makes a VCF file into a seq file, for use with the PHYLIP package.\n",
    "def vcf_to_seqs(invcf,outseq,CHROME='',START=0,STOP=0):\n",
    "    \n",
    "    w = open(outseq,'w')\n",
    "    f = open(invcf,'r')\n",
    "    \n",
    "    for line in f:\n",
    "        line = line.strip()\n",
    "        if line.startswith('##'):\n",
    "            continue\n",
    "        split = line.split()\n",
    "        ids = split[9:]\n",
    "        names = map(PHYLIP_sample_dict.get,ids,ids)\n",
    "        break\n",
    "    print ids,names\n",
    "    f.close()\n",
    "    seqlist = [[] for i in range(len(names))]\n",
    "    var_count = 0\n",
    "    for cur_variant in VCF_parser(invcf):\n",
    "        if CHROME and START and STOP:\n",
    "            \n",
    "            if cur_variant.chrome != CHROME:\n",
    "                continue\n",
    "            if cur_variant.pos < START or cur_variant.pos > STOP:\n",
    "                continue\n",
    "        INDEL = False\n",
    "        for var_seq in cur_variant.nuc_seqs:\n",
    "            if len(var_seq) > 1:\n",
    "               INDEL = True \n",
    "        if INDEL:\n",
    "            continue\n",
    "        ##The dreaded double lambda function. This turns the nested string gts into ints\n",
    "        #genotypes = map(lambda x: map(lambda y: int(y),x),cur_variant.gts)\n",
    "        var_count += 1\n",
    "        for i,geno in enumerate(cur_variant.gts):\n",
    "            if '.' in geno:\n",
    "                seqlist[i].append('N')\n",
    "                continue\n",
    "            geno = map(lambda x: int(x),geno)\n",
    "            seq1 = cur_variant.nuc_seqs[geno[0]]\n",
    "            seq2 = cur_variant.nuc_seqs[geno[1]]\n",
    "            ##Test if the sequences are equal - if not, give the dinucleotide code\n",
    "            if seq1 == seq2:\n",
    "                if seq1 == '*':\n",
    "                    seq1 = 'N'\n",
    "                seqlist[i].append(seq1)\n",
    "            else:\n",
    "                dinuc = convertDiNT(seq1,seq2)\n",
    "                seqlist[i].append(dinuc)\n",
    "            \n",
    "            \n",
    "    \n",
    "    for i in range(len(seqlist)):\n",
    "        seqlist[i] = ''.join(seqlist[i])\n",
    "    \n",
    "    \n",
    "    ##write the seq outfile\n",
    "    header = '  '+str(len(names))+'\\t'+str(var_count)\n",
    "    w.write(header + '\\n')\n",
    "    for i in range(len(names)):\n",
    "        w.write(names[i] + '\\t' + seqlist[i] + '\\n')\n",
    "        \n",
    "    w.flush()\n",
    "    w.close()\n",
    "\n",
    "    \n",
    "    \n",
    "    \n",
    "#vcf_to_gts('MOAV_071817_tested_filtered_haplotype.vcf','MOAV_071817_tested_filtered_haplotype.gt')   \n",
    "#vcf_to_seqs('MOAV_071817_tested_filtered_haplotype.vcf','MOAV_071817_tested_filtered_haplotype.seq')   \n",
    "\n",
    "#vcf_to_gts('MOAV_071817_tested_filtered_haplotype.vcf','MOAV_071817_tested_filtered_haplotype_I4.gt',CHROME='chrXXI',START=3853685,STOP=3858045)\n",
    "#vcf_to_seqs('MOAV_071817_tested_filtered_haplotype.vcf','MOAV_071817_tested_filtered_haplotype_I4.seq',CHROME='chrXXI',START=3853685,STOP=3858045)\n",
    "#vcf_to_seqs('MOAV_071817_tested_haplotype.vcf','MOAV_071817_tested_haplotype_I4.seq',CHROME='chrXXI',START=3853685,STOP=3858045)\n",
    "#vcf_to_seqs('MOAV_071817_tested_haplotype.vcf','MOAV_071817_tested_haplotype_I4_min.seq',CHROME='chrXXI',START=3856000,STOP=3856500)\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
